plugin/evm/block_builder.go

```diff
@@ -111,6 +111,7 @@ func (b *blockBuilder) needToBuild() bool {
 func (b *blockBuilder) markBuilding() {
 	// If the engine has not called BuildBlock, no need to send another message.
 	if b.buildSent {
```

Comment:
Do we want to bound this somehow as a failsafe?

Reply:
Hmm that's an interesting idea. I feel like the invariant should be that if we send `PendingTxs` to consensus, we are guaranteed that `BuildBlock` should be called on the VM.

If we do want to bound this by making this something like an atomic int we can as part of a separate PR but it's not something I'm personally inclined towards since I don't feel like the VM should have to worry about this from its perspective. I feel like I'd rather have either extreme of us only sending the notification once until the `BuildBlock` is called, or just not even track this `builtSent` flag at all.

Reply:
IMO the engine should signal this to the VM (probably with `BuildBlock` taking an extra err or in the block.context) so that VMs can know that proposerVM has failed creating the block. and they can reset `buildSent`.

Or we can set a safe timeout for this (>30 sec) and reset the `buildSent` once we timeout on `BuildBlock`. 

cc @darioush @aaronbuchwald

Reply:
This will currently emit a debug log every time we signal there are new transactions, so each time a new transactions event is issued, which seems more often than we would want imo.